more_itertools/more.py

```diff
@@ -1635,6 +1636,24 @@ def islice_extended(iterable, *args):
                 yield item
 
 
+def reversed_extended(iterable):
```

Comment:
What do you think about `always_reversible` instead of `reversed_extended`?

Reply:
Probably better.